lib/engine/game/base.rb

```diff
@@ -1074,6 +1075,15 @@ def cache_objects
       def bankruptcy_limit_reached?
         @players.any?(&:bankrupt)
       end
+
+      def all_potential_upgrades(tile)
```

Comment:
is this code in tracker?

Reply:
It's very similar, but this doesn't limit on phase or if the tile is available

Reply:
unsure what needs changing if any on this.

Reply:
i'm going to move potential tiles out of step

Reply:
eh maybe later